consensus/src/block_storage/sync_manager.rs

```diff
@@ -469,6 +469,11 @@ impl BlockRetriever {
                 retrieve_batch_size,
                 target_block_id,
             );
+            let self_author = self.network.author();
+            let self_request =
+                self.network
+                    .request_block(request.clone(), self.network.author(), rpc_timeout);
+            futures.push(async move { (self_author, self_request.await) }.boxed());
```

Comment:
The validator has requested the block because it isn't found locally. How does making a self request here benefit us?